great_expectations/metrics/column/column_values_match_regex_values.py

```diff
@@ -0,0 +1,13 @@
+from typing import Union
+
+from great_expectations.metrics.column import ColumnMetric
+from great_expectations.metrics.metric_results import MetricResult
+
+
+class ColumnValuesMatchRegexValuesResult(MetricResult[list[str]]): ...
+
+
+class ColumnValuesMatchRegexValues(ColumnMetric[ColumnValuesMatchRegexValuesResult]):
+    name = "column_values.match_regex"
+    regex: str
+    limit: Union[int, None] = 20
```

Comment:
Maybe I'm opening a can of worms by allowing `None`? IDK, seems reasonable to me...

Reply:
It looks like `None` is not part of the API we are replacing. I think we should not allow `None` and should rename this metric to indicate it will always be a sample of the values. Adding `None` later wouldn't be a breaking change.

Reply:
Works for me! I'll update tomorrow